hack/machineset.sh

```diff
@@ -36,11 +36,24 @@ get_spec() {
   if [ "$byoh" = "true" ]; then
     machineSetName="winbyoh"
   fi
-  # check provider
-  if [ "$provider" = "aws" ] || [ "$provider" = "gce" ]; then
-    # improve machineset name for aws/gcp provider
-    machineSetName="$infraID"-"$machineSetName"-"$az"
-  fi
+  # improve machineset name for supported provider
+  case "$provider" in
+      AWS)
+        machineSetName="$infraID"-"$machineSetName"-"$az"
+      ;;
+      Azure)
+        machineSetName="$infraID"-"$machineSetName"
+      ;;
+      GCP)
+        machineSetName="$infraID"-"$machineSetName"-"$az"
+      ;;
+      VSphere)
+        machineSetName="$infraID"-"$machineSetName"
```

Comment:
not sure this will work for azure and vsphere, see above
```
   # set machineset name, short name for Azure and vSphere due to
   # the limit in the number of characters for VM name
   machineSetName="winworker"
```

Reply:
You are right. Reverted the change. Thanks.